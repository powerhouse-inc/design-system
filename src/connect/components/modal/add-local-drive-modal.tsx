import { AddLocalDriveForm, AddLocalDriveInput, Divider } from '@/connect';
import { DivProps, Icon, Modal } from '@/powerhouse';
import { ComponentPropsWithoutRef } from 'react';
import { twMerge } from 'tailwind-merge';

type ModalProps = ComponentPropsWithoutRef<typeof Modal>;
export type AddLocalDriveModal = {
    open: boolean;
    onOpenChange: (open: boolean) => void;
    onSubmit: (data: AddLocalDriveInput) => void;
    modalProps?: ModalProps;
    containerProps?: DivProps;
};
export function AddLocalDriveModal(props: AddLocalDriveModal) {
    const { open, onOpenChange, onSubmit, modalProps, containerProps } = props;
    function handleCancel() {
        onOpenChange(false);
    }
    return (
        <Modal
            {...modalProps}
            open={open}
            onOpenChange={onOpenChange}
            contentProps={{
                className: 'rounded-2xl',
            }}
        >
            <div
                {...containerProps}
                className={twMerge(
                    'max-w-[408px] rounded-2xl p-6',
                    containerProps?.className,
                )}
            >
                <div className="flex justify-between">
                    <h1 className="text-xl font-bold">Create new drive </h1>
                    <button
                        className="flex size-8 items-center justify-center rounded-md bg-gray-100 text-gray-500 outline-none hover:text-gray-900"
                        onClick={handleCancel}
                        tabIndex={-1}
                    >
                        <Icon name="xmark-light" size={24} />
                    </button>
                </div>
                <Divider className="my-4" />
                <AddLocalDriveForm
                    onSubmit={onSubmit}
                    onCancel={handleCancel}
                />
            </div>
        </Modal>
    );
}
