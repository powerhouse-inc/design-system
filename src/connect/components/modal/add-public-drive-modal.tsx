import { AddPublicDriveForm, Divider } from '@/connect';
import { DivProps, Icon, Modal } from '@/powerhouse';
import { ComponentPropsWithoutRef } from 'react';
import { twMerge } from 'tailwind-merge';

type ModalProps = ComponentPropsWithoutRef<typeof Modal>;
type FormProps = ComponentPropsWithoutRef<typeof AddPublicDriveForm>;
export type AddPublicDriveModalProps = {
    formProps: FormProps;
    modalProps?: ModalProps;
    containerProps?: DivProps;
};
export function AddPublicDriveModal(props: AddPublicDriveModalProps) {
    function handleCancel() {
        props.formProps.onCancel();
        props.modalProps?.onOpenChange?.(false);
    }
    return (
        <Modal
            {...props.modalProps}
            contentProps={{
                className: 'rounded-2xl',
            }}
        >
            <div
                {...props.containerProps}
                className={twMerge(
                    'min-w-[408px] max-w-[408px] rounded-2xl p-6',
                    props.containerProps?.className,
                )}
            >
                <div className="flex justify-between">
                    <h1 className="text-xl font-bold">Add drive</h1>
                    <button
                        className="flex size-8 items-center justify-center rounded-md bg-gray-100 text-gray-500 outline-none hover:text-gray-900"
                        onClick={() => props.modalProps?.onOpenChange?.(false)}
                        tabIndex={-1}
                    >
                        <Icon name="xmark-light" size={24} />
                    </button>
                </div>
                <Divider className="my-4" />
                <AddPublicDriveForm
                    {...props.formProps}
                    onCancel={handleCancel}
                />
            </div>
        </Modal>
    );
}
