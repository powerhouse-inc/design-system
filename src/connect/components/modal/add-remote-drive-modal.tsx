import {
    AddRemoteDriveForm,
    AddRemoteDriveInput,
    Divider,
    SharingType,
} from '@/connect';
import { DivProps, Icon, Modal } from '@/powerhouse';
import { ComponentPropsWithoutRef } from 'react';
import { twMerge } from 'tailwind-merge';

type ModalProps = ComponentPropsWithoutRef<typeof Modal>;
export type AddRemoteDriveModal = {
    open: boolean;
    onOpenChange: (open: boolean) => void;
    onSubmit: (data: AddRemoteDriveInput) => void;
    sharingType: SharingType;
    modalProps?: ModalProps;
    containerProps?: DivProps;
};
export function AddRemoteDriveModal(props: AddRemoteDriveModal) {
    const {
        open,
        onOpenChange,
        onSubmit,
        sharingType,
        modalProps,
        containerProps,
    } = props;
    function handleCancel() {
        onOpenChange(false);
    }
    return (
        <Modal
            {...modalProps}
            open={open}
            onOpenChange={onOpenChange}
            contentProps={{
                className: 'rounded-2xl',
            }}
        >
            <div
                {...containerProps}
                className={twMerge(
                    'min-w-[408px] max-w-[408px] rounded-2xl p-6',
                    containerProps?.className,
                )}
            >
                <div className="flex justify-between">
                    <h1 className="text-xl font-bold">Add drive</h1>
                    <button
                        className="flex size-8 items-center justify-center rounded-md bg-gray-100 text-gray-500 outline-none hover:text-gray-900"
                        onClick={handleCancel}
                        tabIndex={-1}
                    >
                        <Icon name="xmark-light" size={24} />
                    </button>
                </div>
                <Divider className="my-4" />
                <AddRemoteDriveForm
                    sharingType={sharingType}
                    onSubmit={onSubmit}
                    onCancel={handleCancel}
                />
            </div>
        </Modal>
    );
}
