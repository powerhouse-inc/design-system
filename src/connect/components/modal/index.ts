export * from './add-public-drive-modal';
export * from './create-local-drive-modal';
export * from './delete-item-modal';
export * from './drive-settings-modal';
export * from './upgrade-drive-modal';
