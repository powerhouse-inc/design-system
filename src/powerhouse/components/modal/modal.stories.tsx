import { DropdownMenu } from '@/powerhouse';
import { useArgs } from '@storybook/preview-api';
import { Meta, StoryObj } from '@storybook/react';
import { Modal } from '.';

const meta = {
    title: 'Powerhouse/Components/NewModal',
    component: Modal,
    parameters: {
        layout: 'fullscreen',
    },
} satisfies Meta<typeof Modal>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        open: true,
    },
    render: function Wrapper(args) {
        const [, setArgs] = useArgs<typeof args>();

        return (
            <div className="h-screen w-screen bg-white">
                <Modal
                    {...args}
                    open={args.open}
                    onOpenChange={open => {
                        setArgs({ ...args, open });
                    }}
                >
                    <div>
                        <DropdownMenu
                            onItemClick={id => alert(id)}
                            className="mb-3 block items-center justify-center rounded bg-blue-500 text-white"
                            menuClassName="border-2 border-indigo-600 w-64 rounded bg-white cursor-pointer"
                            menuItemClassName="hover:bg-gray-200 px-2"
                            items={[
                                {
                                    id: 'item-1',
                                    content: 'Item 1',
                                },
                                {
                                    id: 'item-2',
                                    content: 'Item 2',
                                },
                                {
                                    id: 'item-3',
                                    content: 'Item 3',
                                },
                            ]}
                        >
                            open menu
                        </DropdownMenu>
                        lorem ipsum dolor sit amet consectetur adipisicing elit.
                        Quisquam quod, voluptatum, quae voluptatem voluptas
                        lorem ipsum dolor sit amet consectetur adipisicing elit.
                        Quisquam quod, voluptatum, quae voluptatem voluptas
                        lorem ipsum dolor sit amet consectetur adipisicing elit.
                        Quisquam quod, voluptatum, quae voluptatem voluptaslorem
                        ipsum dolor sit amet consectetur adipisicing elit.
                        Quisquam quod, voluptatum, quae voluptatem voluptaslorem
                        ipsum dolor sit amet consectetur adipisicing elit.
                        Quisquam quod, voluptatum, quae voluptatem voluptaslorem
                        ipsum dolor sit amet consectetur adipisicing elit.
                        Quisquam quod, voluptatum, quae voluptatem voluptaslorem
                        ipsum dolor sit amet consectetur adipisicing elit.
                        Quisquam quod, voluptatum, quae voluptatem voluptaslorem
                        ipsum dolor sit amet consectetur adipisicing elit.
                        Quisquam quod, voluptatum, quae voluptatem voluptaslorem
                        ipsum dolor sit amet consectetur adipisicing elit.
                        Quisquam quod, voluptatum, quae voluptatem
                        voluptasloremquae voluptatem voluptaslorem ipsum dolor
                        sit amet consectetur adipisicing elit. Quisquam quod,
                        voluptatum, quae voluptatem voluptaslorem ipsum dolor
                        sit amet consectetur adipisicing elit. Quisquam quod,
                        voluptatum, quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptasloremquae voluptatem
                        voluptaslorem ipsum dolor sit amet consectetur
                        adipisicing elit. Quisquam quod, voluptatum, quae
                        voluptatem voluptaslorem ipsum dolor sit amet
                        consectetur adipisicing elit. Quisquam quod, voluptatum,
                        quae voluptatem voluptaslorem
                    </div>
                </Modal>
                <button
                    onClick={() => {
                        setArgs({ ...args, open: true });
                    }}
                >
                    Open modal
                </button>
            </div>
        );
    },
};
