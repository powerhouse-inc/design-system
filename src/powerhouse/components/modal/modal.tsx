import { Content, Overlay, Portal, Root } from '@radix-ui/react-dialog';
import { ComponentPropsWithoutRef } from 'react';
import { twMerge } from 'tailwind-merge';

type Props = {
    children?: React.ReactNode;
    open?: boolean;
    onOpenChange?: (open: boolean) => void;
    overlayProps?: ComponentPropsWithoutRef<typeof Overlay>;
    contentProps?: ComponentPropsWithoutRef<typeof Content>;
};
export function Modal(props: Props) {
    return (
        <Root
            open={props.open}
            defaultOpen={props.open}
            onOpenChange={props.onOpenChange}
        >
            <Portal>
                <Overlay
                    {...props.overlayProps}
                    className={twMerge(
                        'fixed inset-0 grid place-items-center overflow-y-auto bg-slate-900/50 data-[state=closed]:animate-fade-out data-[state=open]:animate-fade-in',
                        props.overlayProps?.className,
                    )}
                >
                    <Content
                        {...props}
                        {...props.contentProps}
                        className={twMerge(
                            'bg-white data-[state=closed]:animate-zoom-out data-[state=open]:animate-zoom-in',
                            props.contentProps?.className,
                        )}
                    >
                        {props.children}
                    </Content>
                </Overlay>
            </Portal>
        </Root>
    );
}
