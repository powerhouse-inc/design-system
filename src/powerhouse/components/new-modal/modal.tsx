import { Content, Overlay, Portal, Root } from '@radix-ui/react-dialog';

type Props = {
    children?: React.ReactNode;
    open?: boolean;
    onOpenChange?: (open: boolean) => void;
};
export function Modal(props: Props) {
    return (
        <Root
            open={props.open}
            defaultOpen={props.open}
            onOpenChange={props.onOpenChange}
        >
            <Portal>
                <Overlay className="fixed inset-0 grid place-items-center overflow-y-auto bg-slate-900/50 data-[state=closed]:animate-fade-out data-[state=open]:animate-fade-in">
                    <Content className="isolate -z-10 min-w-[300px] max-w-[400px] rounded-xl bg-white p-6 data-[state=closed]:animate-zoom-out data-[state=open]:animate-zoom-in">
                        {props.children}
                    </Content>
                </Overlay>
            </Portal>
        </Root>
    );
}
