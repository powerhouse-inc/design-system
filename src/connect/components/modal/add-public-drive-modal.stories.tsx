import { PUBLIC } from '@/connect/constants';
import { useArgs } from '@storybook/preview-api';
import { Meta, StoryObj } from '@storybook/react';
import { AddPublicDriveModal } from './add-public-drive-modal';

const meta = {
    title: 'Connect/Components/Add Public Drive Modal',
    component: AddPublicDriveModal,
} satisfies Meta<typeof AddPublicDriveModal>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        formProps: {
            sharingType: PUBLIC,
            onSubmit() {},
            onCancel() {},
        },
        modalProps: {
            open: true,
        },
    },
    render: function Wrapper(args) {
        const [, setArgs] = useArgs<typeof args>();

        return (
            <div className="grid size-full place-items-center">
                <button
                    className="rounded-lg bg-red-500 p-4 text-white"
                    onClick={() => {
                        setArgs({
                            ...args,
                            modalProps: {
                                ...args.modalProps,
                                open: true,
                            },
                        });
                    }}
                >
                    Open Modal
                </button>
                <AddPublicDriveModal
                    {...args}
                    modalProps={{
                        ...args.modalProps,
                        onOpenChange: open => {
                            setArgs({
                                ...args,
                                modalProps: {
                                    ...args.modalProps,
                                    open,
                                },
                            });
                        },
                    }}
                    formProps={{
                        ...args.formProps,
                        onSubmit: data => {
                            console.log(data);
                            setArgs({
                                ...args,
                                formProps: {
                                    ...args.formProps,
                                    ...data,
                                },
                            });
                        },
                    }}
                />
            </div>
        );
    },
};
