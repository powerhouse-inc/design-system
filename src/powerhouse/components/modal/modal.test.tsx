import { fireEvent, render, screen } from '@testing-library/react';
import { Modal } from './modal';

describe('Modal Component', () => {
    it('should match snapshot', () => {
        render(
            <Modal open={true} onClose={() => {}} data-testid="modal">
                <div>Modal Content</div>
            </Modal>,
        );

        const modalComponent = screen.getByTestId('modal');

        expect(modalComponent).toMatchSnapshot();
    });

    it('should display modal content when open', () => {
        render(
            <Modal open={true} onClose={() => {}}>
                <div>Modal Content</div>
            </Modal>,
        );

        expect(screen.getByText('Modal Content')).toBeInTheDocument();
    });

    it('should call onClose callback when press esc', () => {
        const onClose = vi.fn();
        render(
            <Modal open={true} onClose={onClose}>
                <div>Modal Content</div>
            </Modal>,
        );

        fireEvent.keyDown(screen.getByText('Modal Content'), {
            key: 'Escape',
            code: 'Escape',
        });

        expect(onClose).toHaveBeenCalled();
    });
});
