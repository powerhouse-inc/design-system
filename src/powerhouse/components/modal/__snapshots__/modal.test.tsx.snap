// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal Component > should match snapshot 1`] = `
<div
  aria-describedby="radix-:r2:"
  aria-labelledby="radix-:r1:"
  class="bg-white data-[state=closed]:animate-zoom-out data-[state=open]:animate-zoom-in"
  data-state="open"
  data-testid="modal"
  id="radix-:r0:"
  open=""
  role="dialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <div>
    Modal Content
  </div>
</div>
`;
